opal/mca/accelerator/base/accelerator_base_select.c

```diff
@@ -63,83 +52,32 @@ int opal_accelerator_base_select(void)
         opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
                             "select: initializing %s component %s",
                             component->base_version.mca_type_name,
                             component->base_version.mca_component_name);
-        priority = best_priority;
-        module = component->accelerator_init(&priority);
-        if (NULL == module) {
-            opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
-                                "select: init returned failure for component %s",
-                                component->base_version.mca_component_name);
-            continue;
-        }
-        opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
-                            "select: init returned priority %d", priority);
-        if (priority > best_priority) {
-            best_priority = priority;
-            best_component = component;
-            best_module = module;
-        }
-        aoc = (accelerator_opened_component_t*)malloc(sizeof(accelerator_opened_component_t));
-        if (NULL == aoc) {
-            return OPAL_ERR_OUT_OF_RESOURCE;
-        }
-        OBJ_CONSTRUCT(aoc, opal_list_item_t);
-        aoc->accelerator_component = component;
-        opal_list_append(&opened, (opal_list_item_t*) aoc);
-    }
-
-    /* Finished querying all components. Check for no components found */
 
-    if( NULL == best_component ) {
-        return OPAL_ERR_NOT_FOUND;
-    }
-
-    opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
-                         "selected %s best priority %d\n",
-                         best_component->base_version.mca_component_name, best_priority);
-
-    /* Save the winner */
-    accelerator_base_selected_component = *best_component;
-    opal_accelerator = *best_module;
-
-    /* Finalize all non-selected components */
-
-    for (item = opal_list_remove_first(&opened);
-         NULL != item;
-         item = opal_list_remove_first(&opened)) {
-        aoc = (accelerator_opened_component_t *) item;
-
-        if (aoc->accelerator_component != best_component ) {
-            /* Finalize */
-
-            if (NULL != aoc->accelerator_component->accelerator_finalize) {
-
-                /* Blatently ignore the return code (what would we do to
-                   recover, anyway?  This component is going away, so errors
-                   don't matter anymore) */
-
-                aoc->accelerator_component->accelerator_finalize();
-                opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
-                                    "select: component %s not selected / finalized",
-                                    aoc->accelerator_component->base_version.mca_component_name);
-            }
+        module = component->accelerator_init();
+        if (NULL != module && !found_component) {
```

Comment:
This seems overly complicated, given you're going to essentially abort in one of the two cases.  Why not just have
```
if (NULL != module) {
   if (found_component) {
    opal_show_help(...);
    return NULL;
  }
  found_component = true;
   ....
```

Much cleaner that way.  Also, OMPI style is generally to put the end bracket and the else clause on the same line, so:

```
  } else if (...) {
```
not
```
  }
  else if (...) {
```

Reply:
done